### PR TITLE
feat: Enable AirPlay in MSE

### LINKED ISSUE #5022: Native HLS on Safari only for AirPlay
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**

Yes

**Is your feature request related to a problem? Please describe.**

When using AirPlay on Safari with `useNativeHlsOnSafari: false`, AirPlay becomes unusable. This is because the blob source set on the video element cannot be handled by the AirPlay target.

**Describe the solution you'd like**

As of [Safari 16.4](https://developer.apple.com/documentation/safari-release-notes/safari-16_4-release-notes), Safari can automatically fall back to another supported `<source>` element when starting AirPlay. An example is provided by [this test in the WebKit repository](https://github.com/WebKit/WebKit/commit/4ec10f6ab8204352d7e35c0f16f8219b6688348d). The basic idea would be to provide both the Media Source Extensions-backed video blob and the address of the HLS playlist as separate `<source>` elements as children of the `<video>` element.

**Describe alternatives you've considered**

An alternative would be to re-enable `useNativeHlsOnSafari` when AirPlay is requested by the user, seek to the previous timestamp and start playback again.

### PR DESCRIPTION
**Pull request by tykus160**
Fixes #5022

### CODE REVIEW

**Comment by avelad** [comment]
Webkit issue with the issues we have found: https://bugs.webkit.org/show_bug.cgi?id=281853

**Comment by avelad** [comment]
@shaka-bot tesg

File: lib/player.js [hunk 1]
```diff
@@ -5090,18 +5123,30 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
...
+    };
+    if (this.manifest_ && this.playhead_) {
+      selectMediaSourceMode();
+      // When using MSE + remote we need to set tracks for both MSE and native
+      // apis so that synchronization is maintained.
+      if (!this.isRemotePlayback()) {
+        return;
+      }
     }
+    selectSrcEqualsMode();
```
Comment (joeyparrish) [inline lib/player.js:5149]: Same

File: lib/player.js [hunk 2]
```diff
@@ -5312,20 +5380,32 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
...
+    };
+    if (this.manifest_ && this.playhead_) {
+      selectMediaSourceMode();
+      // When using MSE + remote we need to set tracks for both MSE and native
+      // apis so that synchronization is maintained.
+      if (!this.isRemotePlayback()) {
+        return;
       }
     }
+    selectSrcEqualsMode();
```
Comment (joeyparrish) [inline lib/player.js:5408]: Same

File: lib/player.js [hunk 3]
```diff
@@ -4980,25 +4988,41 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
...
+      selectMediaSourceMode();
+      // When using MSE + remote we need to set tracks for both MSE and native
+      // apis so that synchronization is maintained.
+      if (!this.isRemotePlayback()) {
+        return;
       }
-      this.onTextChanged_();
-      this.setTextDisplayerLanguage_();
     }
+    selectSrcEqualsMode();
```
[the remark this answers is not in the document]
Comment (joeyparrish) [reply lib/player.js:5025]: Ah, okay, there's a comment on 5019-5020 about this.  Disregard.

[the remark this answers is not in the document]
Comment (avelad) [reply lib/player.js:5025]: No, there is an explanation! We need this to sync changes when using AirPlay, when using AirPlay we have src=, but we have to sync changes with our streaming engine :) There is a comment that explains it!

**Review by joeyparrish — APPROVED** [review]

File: lib/player.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (avelad) [reply lib/player.js:5025]: Thanks!

**Pull request merged**